### PR TITLE
Add python example (and change API to support Python 3)

### PR DESCRIPTION
**Pull request by griffin2000**
I added python example to the examples-api-use folder (and edited the API to be python 3 compatible)

### CODE REVIEW

File: examples-api-use/python-example.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+parser.add_argument('-host', help='Server host', default="localhost")
+args = vars(parser.parse_args())
+
+display_width = args["width"]
+display_height = args["height"]
+port = args["port"]
+host = args["host"]
+sq_size = 2
+
+flaschen_conn = Flaschen(host,port,display_width,display_height)
```
Comment (hzeller) [inline examples-api-use/python-example.py:17]: there should be a space after a comma for readability.

File: examples-api-use/python-example.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from flaschen import *
```
Comment (hzeller) [inline examples-api-use/python-example.py:1]: Instead of adding a new file, how about just beefing up the simple-example.py ?

**Comment by hzeller** [comment]
Generally, looks good, but I'd suggest to update the simple-example.py with your change, so that we don't have two essentially same example code snippets.

Also, CC @scottyallen who originally contributed the Python lib.

**Comment by griffin2000** [comment]
Oh I completely missed there was already a python example! Yeah there is definitely no reason to have another. The Python 3 change is still worthwhile integrating, however.

**Comment by hzeller** [comment]
Will the Python3 thing work with both Python2 and 3 ? From what I understand, Python versions are used roughly 50/50

**Comment by griffin2000** [comment]
Yeah it will work in both. There is the odd edge case but most stuff works in both.

File: examples-api-use/flask/static/js/canvas-func.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
+
+
+function init_canvas(w, h)
+{
+
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:5]: Why do you leave multiple lines of whitespace here ?

File: examples-api-use/flask/static/js/canvas-func.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    var ctx = canvas.getContext("2d");
+    var webgl = canvas.getContext("experimental-webgl");
+    function animateFunction(func) {
+        fn = function () {
+            var d = new Date();
+            var t = d.getTime();
+            func(ctx, webgl, t);
+
+            requestAnimationFrame(fn);
+
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:19]: .. or here a free line at the end of a block.

There are more examples below, e.g in animFunc() are two newlines

File: examples-api-use/flask/static/js/canvas-func.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                errMsgDiv.innerHTML = "Error in code:" + e.message
+                return;
+            }
+            funcData.cb = inputUpdateFunc
+            errMsgDiv.innerHTML = "Code is good!"
+        }
+        catch(e)
+        {
+
+
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:81]: here is another example.

File: examples-api-use/flask/static/js/canvas-func.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        ctx.arc(centerX, centerY, radius, 0, 2 * Math.PI, false);
+        ctx.fillStyle = 'green';
+        ctx.fill();
+        ctx.lineWidth = 1;
+        ctx.strokeStyle = '#003300';
+        ctx.stroke();
+    }
+
+
+    function errFunc(ctx, gl, t, w, h) {
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:45]: This is a bit inconsistent: above, you start the brace on a new line, here you put it after the function header. Either is fine, but please stick to one style.

File: examples-api-use/flask/static/js/canvas-func.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        var d = new Date();
+        var t = d.getTime();
+        if (t - lastUpdate < 500)
+            return;
+
+        lastUpdate = t;
+
+        var source = updateFuncInput.value
+        try{
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:65]: Here, you have a try followed by no space and curly brace ...

File: examples-api-use/flask/static/js/canvas-func.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            var inputUpdateFunc = new Function("ctx", "gl", "t", "w", "h", source)
+            try{
+                inputUpdateFunc(ctx,webgl, t)
+            }
+            catch (e) {
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:70]: .... and here you have a space before the curly brace. This should be consistent.
(also should be consistent with whatever you choose with newline-before-curly-brace or brace-in-same-line as discussed above.

File: examples-api-use/flask/static/js/canvas-func.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                funcData.cb = errFunc
+                errMsgDiv.innerHTML = "Error in code:" + e.message
+                return;
+            }
+            funcData.cb = inputUpdateFunc
+            errMsgDiv.innerHTML = "Code is good!"
+        }
+        catch(e)
+        {
```
Comment (hzeller) [inline examples-api-use/flask/static/js/canvas-func.js:79]: e.g. here, you now have a curly brace after a newline. Be consistent.

**Review by hzeller — CHANGES_REQUESTED** [review]
The JavaScript changes are in addition to the Python stuff, ideally it should be a separate pull request. Anyway, now we're here - cool addition, just some code consistency remarks.

File: examples-api-use/flask/ft-flask.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+from flask import *
+from flaschen import *
+app = Flask(__name__)
+
+host = os.getenv("FT_HOST","localhost")
+port = int(os.getenv("FT_PORT","1337"))
+display_width = int(os.getenv("FT_WIDTH","45"))
+display_height = int(os.getenv("FT_HEIGHT","35"))
+
+flaschen_conn = Flaschen(host,port,display_width,display_height, layer=15)
```
Comment (hzeller) [inline examples-api-use/flask/ft-flask.py:14]: there should be a space after each comma.

File: examples-api-use/flask/ft-flask.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import os
+import png
+import base64
+
+from flask import *
+from flaschen import *
+app = Flask(__name__)
+
+host = os.getenv("FT_HOST","localhost")
```
Comment (hzeller) [inline examples-api-use/flask/ft-flask.py:9]: In the other API examples, we use FT_DISPLAY, so we should be consistent here.

**Comment by griffin2000** [comment]
Oh didn't actually know the flask stuff would get submitted. Git snafu on my part. It is totally just hacked together.

**Comment by hzeller** [comment]
Yeah, maybe delete from this pull request and then re-pull-request once it is ready ?

**Pull request closed without merging**